teuthology/orchestra/remote.py

```diff
@@ -401,6 +401,17 @@ def get_file(self, path, sudo=False, dest_dir='/tmp'):
             self.remove(path)
         return local_path
 
+    def write_file(self, path, data):
+        """
+        Write a file on a remote machine.
+
+        :param path: Path on the remote being written to.
+        :param data: Data to be written.
+        """
+        tempfilename = tempfile.mkstemp()[1]
+        self.run(['tee', tempfilename], stdin=data)
+        self.put_file(tempfilename, path)
```

Comment:
It is better to use _putfo_ method of sftp client if you wish to write file over sftp protocol.

Reply:
Why not `put_file()`? I thought it did the same thing.

I can't find `putfo`/`putfoo` method. Are you talking about [`sftp.put()`](https://github.com/ceph/teuthology/blob/master/teuthology/orchestra/remote.py#L310)?

Reply:
@rishabh-d-dave I was talking about "put file object" method _sftp.putfo()_ in opposite to _sftp.put()_, for details look `pydoc paramiko.SFTPClient.putfo`.

Reply:
@rishabh-d-dave and you're trying to use both method of delivering file to a remote host, you have to chose one of: running a tee on a remote host within paramiko.SSHClient.exec_command in the backend, or using paramiko.SFTPClient.putfo in a backend. And as I said it does not require any temporary file creation.

Reply:
>  and you're trying to use both method of delivering file to a remote host, you have to chose one of: running a tee on a remote host within paramiko.SSHClient.exec_command in the backend, or using paramiko.SFTPClient.putfo in a backend. And as I said it does not require any temporary file creation.

Yep, I get it now.